### PR TITLE
Add new strict_visibility_value= attribute to allow exposing strict dependencies to some scopes

### PR DESCRIPTION
**Pull request by dmivankov**
Note that using package default_visibility doesn't always work so explicit visiblity attributes are needed
https://github.com/bazelbuild/bazel/issues/13681

Somewhat related to https://github.com/bazelbuild/rules_jvm_external/pull/648

In combination this for example allows to create a class_name -> maven artifact resolver for
transitive dependency classes while still keeping strict visibility (making transitive dependencies
visible only to the resolve cli tool)

### CODE REVIEW

File: private/rules/maven_install.bzl
Comment (shs96c) [inline private/rules/maven_install.bzl:61]: Nit: "specific" or "specified"

**Review by shs96c — CHANGES_REQUESTED** [review]
That's an amazing Bazel bug. Thanks for spotting this and creating this PR. Could you also please run `./scripts/generate_docs.sh` to make sure that the new flag is properly documented?

**Comment by dmivankov** [comment]
Tests failed, will have a look

File: docs/api.md
Comment (shs96c) [inline docs/api.md:1]: Looks like you don't have doctoc installed: https://github.com/bazelbuild/rules_jvm_external#generating-documentation

**Review by shs96c — APPROVED** [review]
Thank you!

File: docs/api.md
Comment (dmivankov) [reply docs/api.md:1]: oops, didn't notice this in the diff

**Comment by dmivankov** [comment]
funny, cla check thinks @shs96c hasn't signed it even being a codeowner

**Comment by shs96c** [comment]
@dmivankov could you please remove my commit from the history? The CLA bot isn't happy since it came from the GitHub UI and doesn't have my proper email address. I'll merge the PR when it gives me the green light.

**Comment by dmivankov** [comment]
done, squashed and reathored commits related to toc

**Comment by Kernald** [comment]
Because of another Bazel bug (https://github.com/bazelbuild/bazel/issues/13553), accessing the jar files directly was quite useful - see https://github.com/Bencodes/bazel_issue_13553/pull/1. This was possible because of the `default_visibility` at the `package` level, which is now commented out. Is there any way to still access the jar files directly?

**Comment by dmivankov** [comment]
@Kernald jar files should still be accessible, do you have a more detailed repro case?
genrule converting downloaded files (like `@javax_inject_javax_inject_1//file:file`) to `@maven//...` jars should have same visibility as before. You may also try using downloaded jars directly (`bazel query --output build @maven//....jar`)

**Comment by Kernald** [comment]
Such a query doesn't seem to yield anything, but pinning seems to fix the visibility issue. Which seems to point to a slightly different behaviour between 4.3 and 4.4.2 without pinning, but similar behaviour with pinning (and pinning is a totally acceptable solution as far as I'm concerned, I should have done that from the start anyway).

**Pull request merged**
